packages/video_player/video_player_web/lib/src/video_player.dart

```diff
@@ -205,16 +205,27 @@ class VideoPlayer {
   /// Moves the playback head to a new `position`.
   ///
   /// `position` cannot be negative.
+  ///
+  /// Does nothing if the new position is the same as the current position.
   void seekTo(Duration position) {
     assert(!position.isNegative);
+    // Don't seek if video is already at position.
+    // As seeking when completed will trigger another completed event ('onEnded'), causing an infinite loop
+    if (position.inMilliseconds == _getVideoCurrentTimeInMilliseconds()) {
+      return;
+    }
 
     _videoElement.currentTime = position.inMilliseconds.toDouble() / 1000;
   }
 
   /// Returns the current playback head position as a [Duration].
   Duration getPosition() {
     _sendBufferingRangesUpdate();
-    return Duration(milliseconds: (_videoElement.currentTime * 1000).round());
+    return Duration(milliseconds: _getVideoCurrentTimeInMilliseconds());
+  }
+
+  int _getVideoCurrentTimeInMilliseconds() {
```

Comment:
This can use `get` rather than be a method, as it's only doing trivial calculation and has no side effects.